src/tests/integration/D3MCompound.t.sol

```diff
@@ -989,17 +985,72 @@ contract D3MCompoundTest is DSSTest {
         _setRelBorrowTarget(7500);
 
         // Vat is caged for global settlement
-        vat.cage();
+        end.cage();
+        end.cage(ilk);
+        end.skim(ilk, address(d3mCompoundPool));
 
         // Simulate DAI holder gets some gems from GS
-        vat.grab(ilk, address(d3mCompoundPool), address(this), address(this), -int256(100 * 1e18), -int256(0));
+        vm.prank(address(end));
+        vat.flux(ilk, address(end), address(this), 100 * 1e18);
+
+        uint256 totalArt = end.Art(ilk);
+
+        assertEq(cDai.balanceOf(address(this)), 0);
 
         // User can exit and get the cDAI
+        uint256 expectedCdai = 100 * 1e18 * cDai.balanceOf(address(d3mCompoundPool)) / totalArt;
         d3mHub.exit(ilk, address(this), 100 * 1e18);
-
-        uint256 expectedCdai = _wdiv(100 * 1e18, cDai.exchangeRateCurrent());
         assertEq(cDai.balanceOf(address(this)), expectedCdai);
-        assertEqRounding(100 * 1e18, cDai.balanceOfUnderlying(address(this)));
+        assertEqRounding(100 * 1e18, cDai.balanceOfUnderlying(address(this))); // As the whole thing happened in a block (no fees)
+    }
+
+    function test_cage_exit_multiple() public {
+        _setRelBorrowTarget(7500);
+
+        // Vat is caged for global settlement
+        end.cage();
+        end.cage(ilk);
+        end.skim(ilk, address(d3mCompoundPool));
+
+        uint256 totalArt = end.Art(ilk);
+
+        // Simulate DAI holder gets some gems from GS
+        vm.prank(address(end));
+        vat.flux(ilk, address(end), address(this), totalArt);
+
+        uint256 initialCDaiBalance = cDai.balanceOf(address(d3mCompoundPool));
+
+        assertEq(cDai.balanceOf(address(this)), 0);
+
+        // User can exit and get the cDAI
+        uint256 expectedCDai = 25 * 1e18 * initialCDaiBalance / totalArt;
```

Comment:
why switch to `1e18` from `WAD`?

Reply:
That's how it was already in the cDai tests

Reply:
it looked like it was just a few places in those, so i created this to change to WAD for consistency: https://github.com/makerdao/dss-direct-deposit/pull/88

Reply:
Merged!